archivers/star/Portfile

```diff
@@ -1,51 +1,114 @@
+                    BSD tar including a built in 'find', fast fifo, \
+                    selectable cli, accurate sparse files, pattern matcher, sophiticated \
+                    'diff', no name length limitations, the handling of ctimes, atimes \
+                    and mtimes for files, the prevention of the clobbering of more recent \
+                    files during extraction, automatic byte swap (when needed), automatic \
+                    format detect, automatic compression detect, support for ACLs and \
+                    file flags, support for all inode metadata, sophticated error \
+                    control, 'ed' like pattern subsitutions, fast built in -copy mode \
+                    and true incremental dump and restore features. \
+                    \n \
+                    \nExcept for the Schily cpio (or scpio) archiver, all the Schily \
+                    archivers are fully compliant with the POSIX 1003.1-2001 standard! \
+                    \n \
+                    \nThe Schily cpio (or scpio) archiver is fully compliant with the \
+                    previous POSIX 1003.1-1988 standard.  (The cpio archiver \
+                    was dropped from the POSIX 1003.1-2001 standard). \
+                    \n \
+                    \nPLEASE NOTE:  Unlike the pax archiver provided by macOS, \
+                    Schily's pax (or spax) archiver is fully compliant with the POSIX \
+                    1003.1-2001 standard!  In other words, unlike the pax \
+                    archiver provided by macOS, the Schily pax (or spax) archiver will \
+                    actually create and read archives in the POSIX 1003.1-2001 pax \
+                    archive format! \
+                    \n \
+                    \nAlso included is the Schily tartest utility used to test whether a \
+                    tar archive is POSIX compliant and the star_sym utility which \
+                    reads a inode database file from a previous incremental restore via \
+                    the star archiver and prints the database in a human readable format. \
+                    \n \
+                    \nOmitted from the port are the mt, smt and rmt binaries to control \
+                    magnetic tape drives.
+
+homepage            https://cdrtools.sourceforge.net/private/star.html
+master_sites        sourceforge:s-tar
+checksums           sha1    2ee419a1167a54701183b79e75ce3e11de9872a5 \
+                    rmd160  43ef9e17244cbb63da91a1cb56a57d3d8519da30 \
+                    sha256  1b18662ebae6c3509020953d4464e654c7ded4752774de987d8dc307dc43f776 \
+                    size    1154422
 
-depends_build       port:smake
+patchfiles          patch-RULES__rules.prg.diff \
+                    patch-gnutar-manpage.diff
+
+use_bzip2           yes
 
 worksrcdir          star-${version}
 
-use_configure       no
+configure.compiler  macports-clang-14
```

Comment:
It doesn't compile because not all fifo-related stuff is wrapped into `FIFO` macros.

I suggest to fix it on the first place by wrapping it via `#ifdef FIFO` ;)

Reply:
And I have no idea why switch compiler help